src/components/posts/index.js

```diff
@@ -1,27 +1,20 @@
-import { useEffect, useState } from 'react'
 import Post from '../post'
-import { getPosts } from '../../service/apiClient'
-
-const Posts = () => {
-  const [posts, setPosts] = useState([])
-
-  useEffect(() => {
-    getPosts().then(setPosts)
-  }, [])
 
+const Posts = ({ posts }) => {
   return (
     <>
-      {posts.map((post) => {
-        return (
-          <Post
-            key={post.id}
-            name={`${post.author.firstName} ${post.author.lastName}`}
-            date={post.createdAt}
-            content={post.content}
-            comments={post.comments}
-          />
-        )
-      })}
+      {posts &&
```

Comment:
Let's tidy this up by doing the `posts` check before the actual return. Something like:
```js
const Posts = ({posts}) => {
    if(!posts) {
        return <></>
    }
    ...
}
```